src/main/java/org/apache/commons/lang3/exception/ExceptionUtils.java

```diff
@@ -359,14 +359,16 @@ public static String getRootCauseMessage(final Throwable th) {
      * On JDK1.3 and earlier, the cause exception will not be shown
      * unless the specified throwable alters printStackTrace.</p>
      *
-     * @param throwable  the {@code Throwable} to be examined
+     * @param throwable  the {@code Throwable} to be examined, may be null
      * @return the stack trace as generated by the exception's
-     *  {@code printStackTrace(PrintWriter)} method
+     * {@code printStackTrace(PrintWriter)} method, or an empty String if {@code null} input
      */
     public static String getStackTrace(final Throwable throwable) {
         final StringWriter sw = new StringWriter();
         final PrintWriter pw = new PrintWriter(sw, true);
-        throwable.printStackTrace(pw);
+        if (throwable != null) {
+            throwable.printStackTrace(pw);
+        }
```

Comment:
We do not need to create the writers at all. IMO, just guard the method by starting it with:
```
        if (throwable == null) {
            return StringUtils.EMPTY;
        }
```

Reply:
ping.

Reply:
HI @garydgregory 
I don't see the way to just use String and not break the binary.